app/src/main/java/com/example/playlistmaker/data/TracksRepositoryImpl.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.playlistmaker.data
+
+import com.example.playlistmaker.data.dto.TracksSearchRequest
+import com.example.playlistmaker.data.dto.TracksSearchResponse
+import com.example.playlistmaker.domain.api.TracksRepository
+import com.example.playlistmaker.domain.models.FoundTracksInfo
+import com.example.playlistmaker.domain.models.Track
+
+class TracksRepositoryImpl(private val networkClient: NetworkClient): TracksRepository {
+    override fun findTracks(searchString: String): FoundTracksInfo {
+        try {
+            val response = networkClient.doRequest(TracksSearchRequest(searchString))
+            if (response.resultCode == 200) {
```

Comment:
```suggestion
            if (response.isSuccessfull) {
```
Чтобы не хардкодить магические числа в коде